File: Employee_management_system/src/main/java/com/employee/services/EmployeeService.java
package com.employee.services;

import java.util.ArrayList;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import com.employee.models.Employee;
import com.employee.repository.EmployeeRepoImpl;


@Service
public class EmployeeService
{

	@Autowired
	private EmployeeRepoImpl repo;
	
	//List all employees in table

	public ArrayList<Employee> listAllEmployees()
	{
		return (ArrayList<Employee>) repo.findAll();
	}
	
	//find employee using name

	public List<Employee> findEmployeeByName(String name)
	{
	    return repo.findEmployeeByName(name);
	}
	
	//insert an employee into table
	public ResponseEntity<Employee> insertEmployee(Employee employee)
	{
		repo.insert(employee);
		return ResponseEntity.ok(employee);
	}


	//find employee using id

	public Employee findOneEmployee(long id)
	{
		Employee employee=repo.findEmployeeById(id);
				
		
		return employee;
	}
 
	//delete employee using id

		public void deleteEmployeeById(long id)
		{
			Employee employee=repo.findEmployeeById(id);
			repo.delete(employee);
			
		}
		
		//delete all employees

		public void deleteAllEmployees() {
			repo.deleteAll();
		}
		
		//update employee details
	
		public ResponseEntity<Employee> updateEmployee(long id, Employee employee)
		{
			System.out.println("Calling update repo");
			Employee update=repo.updateEmployee(id, employee);

			return ResponseEntity.ok(update);
		}
		
}

File: Employee_management_system/src/main/resources/react-frontend/src/Services/EmployeeService.js
import axios from 'axios'

//API endpoint 
const EMPLOYEE_BASE_REST_API_URL='http://localhost:8081/employees';

class EmployeeService{

    //Axios call to list all the employees
    getAllEmployees()
    {
         return axios.get(EMPLOYEE_BASE_REST_API_URL+"/showAll")
    }

    //Axios call to insert a new employee
    createEmployee(employee)
    {
        return axios.post(EMPLOYEE_BASE_REST_API_URL+"/insertEmployee",employee)
    }

    //Axios call to get an employee by Id
    getEmployeeById(id)
    {
        return axios.get(EMPLOYEE_BASE_REST_API_URL+"/findByID/"+id)
    }

    //Axios call to update employee details
    updateEmployee(id,employee)
    {
        return axios.put(EMPLOYEE_BASE_REST_API_URL+"/update/"+id,employee)
    }

    //Axios call to delete employee by ID
    deleteEmployee(id)
    {
        return axios.delete(EMPLOYEE_BASE_REST_API_URL+"/delete/"+id)
    }
}

export default new EmployeeService();